tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala

```diff
@@ -0,0 +1,148 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package dagr.tasks.vc
+
+import java.nio.file.Path
+
+import dagr.commons.CommonsDef.yieldAndThen
+import dagr.commons.io.Io
+import dagr.core.config.Configuration
+import dagr.core.execsystem.{Cores, Memory, ResourceSet}
+import dagr.core.tasksystem._
+import dagr.tasks.DagrDef.{FilePath, PathToBam, PathToFasta, PathToVcf}
+import dagr.tasks.DataTypes.Vcf
+import htsjdk.samtools.SamReaderFactory
+
+import scala.collection.mutable.ListBuffer
+
+/**
+  * The VarDictJava directory structure is assumed to be as follows:
+  *   - Root directory: /path/to/git/VarDictJava
+  *   - Perl scripts:   /path/to/git/VarDictJava/VarDict
+  *   - Startup script: /path/to/git/VarDictJava/build/install/VarDict/bin/VarDict
+  */
+object VarDictJava extends Configuration {
+  /** Config key for the VarDictJava directory.  This should be the root project directory. */
+  val RootDirConfigKey = "vardictjava.dir"
+
+  /** Path to the VarDictJava directory. */
+  lazy val RootDir = configure[Path](RootDirConfigKey)
+
+  /** Path to the VarDictJava install directory with the VarDictJava startup script. */
+  lazy val VarDictJava = RootDir.resolve("build/install/VarDict/bin/VarDict")
+
+  /** Path to the VarDictJava bin directory with a multitude of perl scripts. */
+  lazy val BinDir = RootDir.resolve("VarDict")
+
+  /** Pulls a sample name out of a BAM file. */
+  private[vc] def extractSampleName(bam: PathToBam) : String = {
+    val in = SamReaderFactory.make().open(bam)
+    yieldAndThen(in.getFileHeader.getReadGroups.iterator().next().getSample) { in.close() }
+  }
+}
+
+/** Runs VarDictJava to produce its intermediate tabular format. */
+private class VarDictJava(tumorBam: PathToBam,
+                          bed: FilePath,
+                          ref: PathToFasta,
+                          tumorName: String,
+                          minimumAf: Double = 0.01,
+                          includeNonPf: Boolean = false,
+                          minimumQuality: Int = 0,
```

Comment:
VarDict's default for this is Q25, so I think we should probably either mirror that here, or preferably make this an Option[Int] with None as the default.